File: src/main/java/com/natia/secretmod/SkyTweaks.java
package com.natia.secretmod;

import com.natia.secretmod.commands.RepartyCommand;
import com.natia.secretmod.config.SkyTweaksCommand;
import com.natia.secretmod.config.SkyTweaksHUD;
import com.natia.secretmod.config.CoreExtension;
import com.natia.secretmod.commands.SavePickupLog;
import com.natia.secretmod.features.altC.QuickTab;
import com.natia.secretmod.hooks.*;
import com.natia.secretmod.extensions.ExtensionList;
import com.natia.secretmod.core.BlockRenderingHook;
import com.natia.secretmod.features.*;
import com.natia.secretmod.gui.GuiHook;
import com.natia.secretmod.networking.ColorText;
import net.minecraftforge.client.ClientCommandHandler;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

@Mod(
        modid = SkyTweaks.MODID,
        name = "SkyTweaks",
        version = SkyTweaks.VERSION,
        clientSideOnly = true
)
public class SkyTweaks {
    public static final String MODID = "skytweaks-mod";
    public static final String VERSION = "v1.0.4";
    public static final boolean IS_UNSTABLE = false;

    public static CoreExtension configHandler;
    @Mod.Instance(MODID)
    public static SkyTweaks INSTANCE;

    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent event)
    {
        configHandler = new CoreExtension();
        ExtensionList.addExtension(configHandler);
        configHandler.updateConfigVariables();
        SecretUtils.addEnchantAdditives();
    }

    @Mod.EventHandler
    public void init(FMLInitializationEvent event)
    {
        /* Minecraft - Hooks */
        MinecraftForge.EVENT_BUS.register(new RenderHook());
        MinecraftForge.EVENT_BUS.register(new TickHook());
        MinecraftForge.EVENT_BUS.register(new BlockRenderingHook());
        MinecraftForge.EVENT_BUS.register(new GuiHook());
        MinecraftForge.EVENT_BUS.register(new GuiScreenHook());
        MinecraftForge.EVENT_BUS.register(new KeyboardInputHook());
        MinecraftForge.EVENT_BUS.register(new ItemPickupHook());
        MinecraftForge.EVENT_BUS.register(new MessageHook());
        MinecraftForge.EVENT_BUS.register(new WorldHook());

        /* Only do this for things that only have 1 event. */
        MinecraftForge.EVENT_BUS.register(DamagePerSecond.getInstance());
        MinecraftForge.EVENT_BUS.register(new QuickTab());

        /* Cosmetics */
        MinecraftForge.EVENT_BUS.register(new ColorText());

        System.out.println("SkyTweaks Mod Initialized");
    }

    @Mod.EventHandler
    public void postinit(FMLPostInitializationEvent event) {
        ClientCommandHandler.instance.registerCommand(new SavePickupLog());
        ClientCommandHandler.instance.registerCommand(new SkyTweaksCommand());
        ClientCommandHandler.instance.registerCommand(new SkyTweaksHUD());
        ClientCommandHandler.instance.registerCommand(new RepartyCommand());

        configHandler.saveConfig();

        new Thread(SecretUtils::updateBazaarCache).start();

    }
}

File: src/main/kotlin/natia/skytweaks/SkyTweaks.kt
package natia.skytweaks

import gg.blendingMC.BlendingMC
import gg.blendingMC.settings.BlendingMCSettings
import natia.skytweaks.commands.RepartyCommand
import natia.skytweaks.commands.SavePickupLog
import natia.skytweaks.config.SkyTweaksBlend
import natia.skytweaks.config.SkyTweaksCommand
import natia.skytweaks.config.SkyTweaksHUDCommand
import natia.skytweaks.core.BlockRenderingHook
import natia.skytweaks.features.BatDied
import natia.skytweaks.features.DamagePerSecond
import natia.skytweaks.features.altC.QuickTab
import natia.skytweaks.hooks.*
import natia.skytweaks.features.griffin.GriffinBurrowWaypoints
import natia.skytweaks.gui.GuiHook
import natia.skytweaks.networking.ColorText
import net.minecraft.client.Minecraft
import net.minecraftforge.client.ClientCommandHandler
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

@Mod(modid = SkyTweaks.MODID, name = "SkyTweaks", version = SkyTweaks.VERSION, clientSideOnly = true)
class SkyTweaks {

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        BlendingMCSettings.welcomed = true
        BlendingMC.getInstance().fmlinitialize()
        configHandler = SkyTweaksBlend()
        configHandler!!.updateConfigVariables()
        SecretUtils.addEnchantAdditives()
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {
        /* Minecraft - Hooks */
        MinecraftForge.EVENT_BUS.register(RenderHook())
        MinecraftForge.EVENT_BUS.register(TickHook())
        MinecraftForge.EVENT_BUS.register(BlockRenderingHook())
        MinecraftForge.EVENT_BUS.register(GuiHook())

        MinecraftForge.EVENT_BUS.register(GuiScreenHook())
        MinecraftForge.EVENT_BUS.register(KeyboardInputHook())
        MinecraftForge.EVENT_BUS.register(ItemPickupHook())
        MinecraftForge.EVENT_BUS.register(MessageHook())
        MinecraftForge.EVENT_BUS.register(WorldHook())

        /* Only do this for things that only have 1 event. */
        MinecraftForge.EVENT_BUS.register(DamagePerSecond.instance)
        MinecraftForge.EVENT_BUS.register(QuickTab())
        MinecraftForge.EVENT_BUS.register(GriffinBurrowWaypoints.instance)
        MinecraftForge.EVENT_BUS.register(BatDied.instance)

        /* Cosmetics */
        MinecraftForge.EVENT_BUS.register(ColorText())
        LOGGER.info("SkyTweaks Mod Initialized")
    }

    @Mod.EventHandler
    fun postinit(event: FMLPostInitializationEvent) {
        ClientCommandHandler.instance.registerCommand(SavePickupLog())
        ClientCommandHandler.instance.registerCommand(SkyTweaksHUDCommand())
        ClientCommandHandler.instance.registerCommand(SkyTweaksCommand())
        ClientCommandHandler.instance.registerCommand(RepartyCommand())

        configHandler!!.saveConfig()

        Thread(Runnable { SecretUtils.updateBazaarCache() }).start()

    }

    companion object {
        const val MODID: String = "skytweaks-mod"
        const val VERSION = "v1.0.5"
        const val IS_UNSTABLE = false
        @JvmField
        val LOGGER: Logger = LogManager.getLogger("SkyTweaks Logger")!!


        var configHandler: SkyTweaksBlend? = null

        @Mod.Instance(MODID)
        var INSTANCE: SkyTweaks? = null
    }
}
